File: Learn_Android/Old/getWallpaer/app/src/main/java/com/example/sunkai/getwallpaer/Activity/MainActivity.java
package com.example.sunkai.getwallpaer.Activity;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;

import com.example.sunkai.getwallpaer.Adapter.RecyclerViewAdapter;
import com.example.sunkai.getwallpaer.ConenctService.ConnectImageServiceBuilder;
import com.example.sunkai.getwallpaer.Data.imageData;
import com.example.sunkai.getwallpaer.R;
import com.example.sunkai.getwallpaer.Tools.MakeToast;
import com.example.sunkai.getwallpaer.Tools.Values;

import java.util.List;
import java.util.stream.Collector;

public class MainActivity extends AppCompatActivity {
    private List<imageData> datas;
    private RecyclerView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
        getImageInformations();
    }

    private void initView() {
        recyclerView = findViewById(R.id.recycler_view);
    }

    private void getImageInformations() {
        new Thread(() -> {
            try {
                datas = ConnectImageServiceBuilder.create(Values.IMAGE_URL).post();
                for (imageData data : datas) {
                    if (data.getUrl().equals("20130312")) {
                        data.setUrl(data.getUrl() + "!");
                    }
                }
                datas.stream()
                        .filter(data->data.getUrl().equals("20130312"))
                        .map(data->data.getUrl()+"!")
                        .collect(Collector.of()))
                runOnUiThread(() -> {
                    RecyclerViewAdapter adapter = new RecyclerViewAdapter(MainActivity.this, datas);
                    recyclerView.setLayoutManager(new GridLayoutManager(MainActivity.this, 2));
                    recyclerView.setAdapter(adapter);
                    MakeToast.Toast("成功");
                });
            } catch (Exception e) {
                e.printStackTrace();
            }
        }).start();
    }
}

File: Learn_Android/MyApplication/app/src/main/java/com/example/sunkai/myapplication/MainActivity.kt
package com.example.sunkai.myapplication

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.GridLayoutManager
import android.view.View
import android.widget.AdapterView
import com.example.sunkai.myapplication.Adapter.RecyclerViewAdapter
import com.example.sunkai.myapplication.ConnectWebService.GetImageData
import com.example.sunkai.myapplication.Data.IMAGE_URL
import com.example.sunkai.myapplication.Interface.itemClick
import com.example.sunkai.myapplication.Tools.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(),View.OnClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        getInformation()
    }
    private fun getInformation(){
        Thread{
            val datas= GetImageData(IMAGE_URL)
            runOnUiThread {
                val adpter=RecyclerViewAdapter(this@MainActivity,datas)
                Toast("刷新完成")
                recycler_view.layoutManager=GridLayoutManager(this,2)
                recycler_view.adapter=adpter
            }
        }.start()
    }

    override fun onClick(v: View?) {

    }
}
